feat(lambda): added new property allowAllIpv6Outbound to FunctionOptions (#31013)
aws/aws-cdk: packages/aws-cdk-lib/aws-lambda/README.md

});
```

> ## Outbound traffic
> By default, when creating a Lambda function, it would add a security group outbound rule to allow sending all network traffic (except IPv6). This is controlled by `allowAllOutbound` in function properties, which has a default value of `true`.
> 
> To allow outbound IPv6 traffic by default, explicitly set `allowAllIpv6Outbound` to `true` in function properties as shown below (the default value for `allowAllIpv6Outbound` is `false`):
> ```ts

Reviewer: ```suggestion
## Outbound traffic

By default, when creating a Lambda function, it would add a security group outbound rule to allow sending all network traffic (except IPv6).
You can override the default behavior by setting the `allowAllOutbound` property to `false`.

To allow outbound IPv6 traffic by default, explicitly set the `allowAllIpv6Outbound` property to `true`, as shown below.
The default value for `allowAllIpv6Outbound` is `false`.

```ts
```

});
```

## Outbound traffic
By default, when creating a Lambda function, it would add a security group outbound rule to allow sending all network traffic (except IPv6). This is controlled by `allowAllOutbound` in function properties, which has a default value of `true`.

To allow outbound IPv6 traffic by default, explicitly set `allowAllIpv6Outbound` to `true` in function properties as shown below (the default value for `allowAllIpv6Outbound` is `false`):
```ts
const vpc = new ec2.Vpc(this, 'Vpc', {
  maxAzs: 2,

>   restrictDefaultSecurityGroup: false,

Reviewer 2: Is `restrictDefaultSecurityGroup: false` required if we need to allow all outbound traffic? If yes, maybe we should note this in the README? If not, let's remove this from the sample?

Author: Removed

> const vpc = new ec2.Vpc(this, 'Vpc', {
>   maxAzs: 2,
>   restrictDefaultSecurityGroup: false,
> });

Reviewer 2: ```suggestion
import * as ec2 from 'aws-cdk-lib/aws-ec2';

const vpc = new ec2.Vpc(this, 'Vpc', {
  maxAzs: 2,
  restrictDefaultSecurityGroup: false,
});
```

Author: Fixed

});
```

## Outbound traffic
By default, when creating a Lambda function, it would add a security group outbound rule to allow sending all network traffic (except IPv6). This is controlled by `allowAllOutbound` in function properties, which has a default value of `true`.

To allow outbound IPv6 traffic by default, explicitly set `allowAllIpv6Outbound` to `true` in function properties as shown below (the default value for `allowAllIpv6Outbound` is `false`):
```ts
import * as ec2 from 'aws-cdk-lib/aws-ec2';

> const vpc = new ec2.Vpc(this, 'Vpc', {
>   maxAzs: 2,
> });

Reviewer 2: ```suggestion
const vpc = new ec2.Vpc(this, 'Vpc');
```

Let's just remove optional props and keep the sample as simple as possible.

Author: Done

});
```

> To allow outbound IPv6 traffic by default, set `allowAllIpv6Outbound` to true.

Reviewer: What about defining a `## Outbound traffic` section that explains the behavior for both `allowAllOutbound` and `allowAllIpv6Outbound`?

Author: Updated. Thanks for the suggestion.
